src/idl_parser.cpp

```diff
@@ -2579,6 +2585,112 @@ void Parser::Serialize() {
   }
 }
 
+static Namespace *GetNamespace(
+    const std::string &qualified_name, std::vector<Namespace *> &namespaces,
+    std::map<std::string, Namespace *> &namespaces_index) {
+  const char *dot = strrchr(qualified_name.c_str(), '.');
+  std::string namespace_name =
+      dot ? namespace_name.assign(qualified_name.c_str(), dot) : "";
```

Comment:
using namespace_name to construct itself?

Reply:
No the qualified name is the qualified type name.  The namespace name is everything before the last dot.